src/main/java/org/folio/search/service/ResourceIdService.java

```diff
@@ -138,16 +138,23 @@ protected OutputStreamWriter createOutputStreamWriter(OutputStream outputStream)
   }
 
   private void streamResourceIds(CqlResourceIdsRequest request, Consumer<List<String>> idsConsumer) {
-    log.info("streamResourceIds:: by [query: {}, resource: {}]", request.getQuery(), request.getResource());
+    boolean notSharedActiveAffiliation = !request.getTenantId().equalsIgnoreCase("consortium")
+      && request.getQuery().contains("shared==\"false\"");
 
-    var searchSource = queryConverter.convertForConsortia(request.getQuery(), request.getResource())
+    var searchSource = getSearchSource(request, notSharedActiveAffiliation)
       .size(streamIdsProperties.getScrollQuerySize())
       .fetchSource(new String[] {request.getSourceFieldPath()}, null)
       .sort(fieldSort("_doc"));
 
     searchRepository.streamResourceIds(request, searchSource, idsConsumer);
   }
 
+  private SearchSourceBuilder getSearchSource(CqlResourceIdsRequest req, boolean notSharedActiveAffiliation) {
+    return notSharedActiveAffiliation
+      ? queryConverter.convertNotSharedActiveAffiliation(req.getQuery(), req.getResource(), req.getTenantId())
+      : queryConverter.convertForConsortia(req.getQuery(), req.getResource());
+  }
+
```

Comment:
I suggest to not do a check and always use `queryConverter.convertForConsortia(req.getQuery(), req.getResource())`

Reply:
done